### PR TITLE
feat(SF-998): Add mixin support

### PR DESCRIPTION
**Pull request by Vic-Dev**

### CODE REVIEW

File: src/core/configuration.ts [hunk 1]
```diff
@@ -22,11 +23,16 @@ interface Configuration extends FluxConfiguration {
     globalMixin?: boolean;
     riot?: any;
   };
+
+  mixins?: {
+    global?: GlobalMixin;
+  } & {
```
Comment (jmbtutor) [inline src/core/configuration.ts]: We can merge these two and remove this line.

File: test/unit/core/system.ts [hunk 2]
```diff
@@ -41,13 +42,24 @@ suite('System', ({ expect, spy, stub }) => {
...
+      sinon.assert.callOrder(
+        initConfig,
+        initRiot,
+        initFlux,
+        createServices,
+        initServices,
+        initMixin,
+        registerTags,
+        initUserMixins
+      );
```
Comment (jmbtutor) [inline test/unit/core/system.ts]: We could potentially create an array of the functions in order, loop through it asserting that each one is called, then assert that `initConfig` and `createServices` are called with the proper arguments, then destructure the array in `callOrder`.

File: test/unit/core/system.ts [hunk 3]
```diff
@@ -233,6 +245,52 @@ suite('System', ({ expect, spy, stub }) => {
...
+        },
+      };
+      const system = new System(app);
+
+      system.initUserMixins();
+
+      expect(mixin).to.be.calledWith(globalMixin);
+    });
+
+    it('should make other mixins available', () => {
```
Comment (jmbtutor) [inline test/unit/core/system.ts]: s/other/custom/ : "should make **custom** mixins available"

File: test/unit/core/system.ts (hunk 2 above)
Comment (Vic-Dev) [reply test/unit/core/system.ts]: I think it's fine

**Pull request merged**